### ISSUE TITLE
-Wformat-truncation warnings in vrtpprintf

### ISSUE THREAD

**Issue by xavierba** [coturn/coturn#375]
When building on Fedora, I get several -Wformat-truncation warnings in the vrtpprintf function from src/apps/common/ns_turn_utils.c
Can you please take a look ?
```
BUILDSTDERR: src/apps/common/ns_turn_utils.c: In function 'vrtpprintf':
BUILDSTDERR: src/apps/common/ns_turn_utils.c:396:56: warning: '%s' directive output may be truncated writing up to 1024 bytes into a region of size 1005 [-Wformat-truncation=]
BUILDSTDERR:    snprintf(logbase, FILE_STR_LEN, "/var/log/turnserver/%s", logtail);
BUILDSTDERR:                                                         ^~   ~~~~~~~
BUILDSTDERR: In file included from /usr/include/stdio.h:873,
BUILDSTDERR:                  from src/ns_turn_defs.h:49,
BUILDSTDERR:                  from src/client/ns_turn_ioaddr.h:34,
BUILDSTDERR:                  from src/apps/common/ns_turn_utils.h:39,
BUILDSTDERR:                  from src/apps/common/ns_turn_utils.c:31:
BUILDSTDERR: /usr/include/bits/stdio2.h:67:10: note: '__builtin___snprintf_chk' output between 21 and 1045 bytes into a destination of size 1025
BUILDSTDERR:    return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
BUILDSTDERR:           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR:         __bos (__s), __fmt, __va_arg_pack ());
BUILDSTDERR:         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR: src/apps/common/ns_turn_utils.c:404:46: warning: '%s' directive output may be truncated writing up to 1024 bytes into a region of size 1016 [-Wformat-truncation=]
BUILDSTDERR:     snprintf(logbase, FILE_STR_LEN, "/var/log/%s", logtail);
BUILDSTDERR:                                               ^~   ~~~~~~~
BUILDSTDERR: In file included from /usr/include/stdio.h:873,
BUILDSTDERR:                  from src/ns_turn_defs.h:49,
BUILDSTDERR:                  from src/client/ns_turn_ioaddr.h:34,
BUILDSTDERR:                  from src/apps/common/ns_turn_utils.h:39,
BUILDSTDERR:                  from src/apps/common/ns_turn_utils.c:31:
BUILDSTDERR: /usr/include/bits/stdio2.h:67:10: note: '__builtin___snprintf_chk' output between 10 and 1034 bytes into a destination of size 1025
BUILDSTDERR:    return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
BUILDSTDERR:           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR:         __bos (__s), __fmt, __va_arg_pack ());
BUILDSTDERR:         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR: src/apps/common/ns_turn_utils.c:411:47: warning: '%s' directive output may be truncated writing up to 1024 bytes into a region of size 1016 [-Wformat-truncation=]
BUILDSTDERR:      snprintf(logbase, FILE_STR_LEN, "/var/tmp/%s", logtail);
BUILDSTDERR:                                                ^~   ~~~~~~~
BUILDSTDERR: In file included from /usr/include/stdio.h:873,
BUILDSTDERR:                  from src/ns_turn_defs.h:49,
BUILDSTDERR:                  from src/client/ns_turn_ioaddr.h:34,
BUILDSTDERR:                  from src/apps/common/ns_turn_utils.h:39,
BUILDSTDERR:                  from src/apps/common/ns_turn_utils.c:31:
BUILDSTDERR: /usr/include/bits/stdio2.h:67:10: note: '__builtin___snprintf_chk' output between 10 and 1034 bytes into a destination of size 1025
BUILDSTDERR:    return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
BUILDSTDERR:           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR:         __bos (__s), __fmt, __va_arg_pack ());
BUILDSTDERR:         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR: src/apps/common/ns_turn_utils.c:417:44: warning: '%s' directive output may be truncated writing up to 1024 bytes into a region of size 1020 [-Wformat-truncation=]
BUILDSTDERR:       snprintf(logbase, FILE_STR_LEN, "/tmp/%s", logtail);
BUILDSTDERR:                                             ^~   ~~~~~~~
BUILDSTDERR: In file included from /usr/include/stdio.h:873,
BUILDSTDERR:                  from src/ns_turn_defs.h:49,
BUILDSTDERR:                  from src/client/ns_turn_ioaddr.h:34,
BUILDSTDERR:                  from src/apps/common/ns_turn_utils.h:39,
BUILDSTDERR:                  from src/apps/common/ns_turn_utils.c:31:
BUILDSTDERR: /usr/include/bits/stdio2.h:67:10: note: '__builtin___snprintf_chk' output between 6 and 1030 bytes into a destination of size 1025
BUILDSTDERR:    return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
BUILDSTDERR:           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR:         __bos (__s), __fmt, __va_arg_pack ());
BUILDSTDERR:         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by misi** [comment]
Which version or git tag?

**Comment by xavierba** [comment]
Sorry, latest stable, 4.5.1.1.

**Comment by eakraly** [comment]
Most probably addressed in 5e87c444693b39e11cabc498450dacbef30ec64a
In any case, `vrtpprintf` is not used in code anymore
